src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1225,6 +1225,87 @@ module T = struct
   let dummy_transaction_pool_proxy : transaction_pool_proxy =
     { find_by_hash = const None }
 
+  let extract_zkapp_veirfication_keys
+      ({ account_updates; _ } : Zkapp_command.Verifiable.t) : _ Result.t =
+    account_updates |> Zkapp_statement.zkapp_statements_of_forest'
+    |> Zkapp_command.Call_forest.With_hashes_and_data
+       .to_zkapp_command_with_hashes_list
+    |> List.fold_result ~init:[]
+         ~f:(fun
+              acc
+              (((p, (vk_opt, _stmt)), _at_account_update) :
+                (Mina_base.Account_update.t * (_ With_hash.t option * _)) * _ )
```

Comment:
Do we need this type annotation? If yes, lets annotate `(p : Mina_base.Account_update.t)` and `(vk_opt : _ With_hash.t option)`. I.e. inline type annotation for the parameters that we use, not for the whole argument.